File: src/main/java/view/PickPawnInput.java
package view;

import org.jline.terminal.Terminal;

import java.io.IOException;
import java.io.InputStream;
import java.io.PrintStream;

class PickPawnInput {
    private Terminal mTerminal;
    private int mSelectedPawnIndex;

    PickPawnInput(Terminal terminal) {
        mTerminal = terminal;
        mSelectedPawnIndex = 0;
    }

    // commands
    void call() throws IOException, InterruptedException {
        mTerminal.writer().println();
        mTerminal.writer().print("select a pawn [< > enter]");

        int code = mTerminal.reader().read();
        switch(code) {
            case 37:
                mSelectedPawnIndex--;
                break;
            case 39:
                mSelectedPawnIndex++;
                break;
            default:
//                    mSelectedPawnIndex = null;
                break;
        }

        mTerminal.writer().println();
    }

    // queries
    Integer getSelectedPawnIndex() {
        return mSelectedPawnIndex;
    }
}

File: Sources/Onitama/view/PIckPawnInput.swift
final class PickPawnInput {
  private let mTerminal:          Terminal
  private var mSelectedPawnIndex: Int
  private var mIsConfirmed:       Bool = false

  // lifetime
  init(terminal: Terminal) {
    mTerminal          = terminal
    mSelectedPawnIndex = 0
  }

  // command
  func call() {
    mTerminal.write("\nselect a pawn [< > enter]")

    switch mTerminal.read() {
    case 37:
      mSelectedPawnIndex -= 1
    case 39:
      mSelectedPawnIndex += 1
    default:
      break
    }

    mTerminal.write("\n")
  }

  // queries
  var selectedPawnIndex: Int? {
    if mIsConfirmed {
      return mSelectedPawnIndex
    } else {
      return nil
    }
  }
}
